torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -85,6 +85,25 @@ bool isInputNonSizeZeroTensor(const Node* node) {
   return true;
 }
 
+bool isScalarTypeCompatible(const Node* node, size_t offset) {
+  auto val = node->input(offset);
+  // return true if it's not specified
+  if (val->type()->isSubtypeOf(static_cast<c10::TypePtr>(NoneType::get()))) {
+    return true;
+  }
+  // return falss if it's runtime value
+  if (val->node()->kind() != prim::Constant) {
+    return false;
+  }
+  auto dtype = toIValue(val)->toScalarType();
+
+  // we do NOT support half math type yet
+  if (dtype == at::ScalarType::Half || dtype == at::ScalarType::BFloat16) {
+    return false;
```

Comment:
We want to avoid merging explicit reduced precision math type cc'ing @kevinstephano